src/rhp.rs

```diff
@@ -12,21 +13,63 @@ pub fn sector_root(sector: &[u8]) -> Hash256 {
     let mut params = Params::new();
     params.hash_length(32);
 
-    let mut tree_hashes = sector
-        .par_chunks_exact(SEGMENT_SIZE)
-        .map(|chunk| sum_leaf(&params, chunk))
-        .collect::<Vec<_>>();
+    let mut tree_hashes = vec![[0; 32]; SECTOR_SIZE / SEGMENT_SIZE];
+    tree_hashes
+        .par_chunks_exact_mut(4)
+        .enumerate()
+        .for_each(|(i, chunk)| {
+            // prepare inputs
+            let mut inputs = [[0u8; SEGMENT_SIZE + 1]; 4];
+            for (j, input) in inputs.iter_mut().enumerate() {
+                input[0] = LEAF_HASH_PREFIX[0];
+                input[1..]
+                    .copy_from_slice(&sector[SEGMENT_SIZE * (i + j)..SEGMENT_SIZE * (i + j + 1)]);
+            }
 
-    let mut chunk_size = 2;
+            // hash them
+            let mut jobs = [
+                HashManyJob::new(&params, &inputs[0][..]),
+                HashManyJob::new(&params, &inputs[1][..]),
```

Comment:
I'm confused. It seems like you're iterating over groups of 4 segments at a time, but here you're only hashing 2 of them? 🤔